src/lazy/any_encoding.rs

```diff
@@ -280,6 +384,13 @@ pub struct LazyRawAnyValue<'top> {
     encoding: LazyRawValueKind<'top>,
 }
 
+impl<'top> LazyRawAnyValue<'top> {
+    // TODO: feature gate
+    pub fn kind(&self) -> LazyRawValueKind<'top> {
+        self.encoding
+    }
+}
```

Comment:
🗺️ Allows an application to access the encoding-specific data type that underlies this value.